### PR TITLE
Cleanup CompatUtility class and features

### PR DESCRIPTION
**Pull request by dnsmichi**

### CODE REVIEW

File: lib/db_ido/hostdbobject.cpp [hunk 1]
```diff
@@ -47,65 +48,53 @@ Dictionary::Ptr HostDbObject::GetConfigFields(void) const
 	Dictionary::Ptr fields = new Dictionary();
 	Host::Ptr host = static_pointer_cast<Host>(GetObject());
 
-	fields->Set("alias", CompatUtility::GetHostAlias(host));
-	fields->Set("display_name", host->GetDisplayName());
+	/* Compatibility fallback. */
+	String displayName = host->GetDisplayName();
+
+	if (!displayName.IsEmpty()) {
```
Comment (gunnarbeutner) [inline lib/db_ido/hostdbobject.cpp]: The braces aren't really necessary here.

File: lib/livestatus/contactstable.cpp [hunk 2]
```diff
@@ -207,7 +207,7 @@ Value ContactsTable::CustomVariableNamesAccessor(const Value& row)
 
 	{
 		ObjectLock olock(user);
-		vars = CompatUtility::GetCustomAttributeConfig(user);
```
Comment (gunnarbeutner) [inline lib/livestatus/contactstable.cpp]: This lock isn't necessary.

File: lib/livestatus/commandstable.cpp [hunk 3]
```diff
@@ -161,7 +161,7 @@ Value CommandsTable::CustomVariablesAccessor(const Value& row)
 
 	{
 		ObjectLock olock(command);
```
Comment (gunnarbeutner) [inline lib/livestatus/commandstable.cpp]: This lock isn't necessary.

File: lib/livestatus/commandstable.cpp [hunk 4]
```diff
@@ -132,7 +132,7 @@ Value CommandsTable::CustomVariableValuesAccessor(const Value& row)
...
```
Comment (gunnarbeutner) [inline lib/livestatus/commandstable.cpp]: This lock isn't necessary.

File: lib/livestatus/commandstable.cpp [hunk 5]
```diff
@@ -103,7 +103,7 @@ Value CommandsTable::CustomVariableNamesAccessor(const Value& row)
...
```
Comment (gunnarbeutner) [inline lib/livestatus/commandstable.cpp]: This lock isn't necessary.

File: lib/livestatus/contactstable.cpp [hunk 6]
```diff
@@ -234,7 +234,7 @@ Value ContactsTable::CustomVariableValuesAccessor(const Value& row)
 
 	{
 		ObjectLock olock(user);
```
Comment (gunnarbeutner) [inline lib/livestatus/contactstable.cpp]: This lock isn't necessary.

File: lib/livestatus/contactstable.cpp [hunk 7]
```diff
@@ -264,7 +264,7 @@ Value ContactsTable::CustomVariablesAccessor(const Value& row)
...
```
Comment (gunnarbeutner) [inline lib/livestatus/contactstable.cpp]: This lock isn't necessary.

File: lib/livestatus/contactstable.cpp [hunk 8]
```diff
@@ -299,7 +299,7 @@ Value ContactsTable::CVIsJsonAccessor(const Value& row)
...
```
Comment (gunnarbeutner) [inline lib/livestatus/contactstable.cpp]: This lock isn't necessary.

**Comment by dnsmichi** [comment]
@gunnarbeutner applied, thx.

**Comment by dnsmichi** [comment]
Rebased against master.

**Pull request merged**